public/js/app/leaderboards.js

```diff
@@ -1,48 +1,36 @@
 //Variables used across the board
-let plusOrMinusPlayerList = 0;
-let addNextPlayer = 0;
-let previousMissingPlayer = 0;
 let pageNumber = 0;
-let lastPage = 1;
-let d = new Date();
-let minusMonth = d.setMonth(d.getMonth() - 1);
-let currentDate = new Date(minusMonth).toISOString();
-let currentLeaderboard = '';
+let lastPage = 5;
+let playerList = [];
+let timedOutPlayers = [];
+let currentLeaderboard = '1v1';
 let backgroundColor = 'greyLeaderboard';
-let nextButton = document.getElementById('nextPage');
-let previousButton = document.getElementById('previousPage');
-let lastButton = document.getElementById('lastPage');
-let firstButton = document.getElementById('firstPage');
+// This decides the time filter
+let d = new Date();
+let timeFilter = 6;
+let minusTimeFilter = d.setMonth(d.getMonth() - timeFilter);
+let currentDate = new Date(minusTimeFilter).toISOString();
+
 
+
+//Names of buttons
 async function leaderboardOneJSON(leaderboardFile) {
   //Check which category is active
   const response = await fetch(`js/app/members/${leaderboardFile}.json`);
-  const data = await response.json();
   currentLeaderboard = leaderboardFile;
+  const data =  response.json();
   return await data;
 }
 
 //Updates the leaderboard according to what is needed
-function leaderboardUpdate(data) {
-
-  // We convert data into a string to find out how many pages we have. We can find so by checking the first two digits. In other words, if we have 1349 players, then we have 13 pages. However, if we have 834, we have 8 pages (only take the first digit).
-
-  let missingPlayers = 0;
-  for (let i = 0; i < data.length; i++) {
-    if (currentDate > data[i].date) {
-      missingPlayers++;
-      
-    }
-  }
-  //console.log(`We are missing ${missingPlayers} players`);
-  let dataMinusMissingPlayer = (data.length - missingPlayers).toString();
-
-  if (dataMinusMissingPlayer.length !== 4) {
-    lastPage = dataMinusMissingPlayer.slice(0, 1);
+function leaderboardUpdate() {
+  // We convert playerList into a string to find out how many pages we have. We can find so by checking the first two digits. In other words, if we have 1349 players, then we have 13 pages. However, if we have 834, we have 8 pages (only take the first digit).
```

Comment:
remove all of this and just divide by 100, then floor. This is how you get the "first two digits" of a number... 😅

Reply:
This was very helpful, thanks!